src/DiskIO/IpcIo/IpcIoFile.cc

```diff
@@ -500,22 +499,25 @@ IpcIoFile::HandleStrandBusyResponse(const Ipc::StrandMessage &response)
 }
 
 void
-IpcIoFile::indexingCompleted()
+IpcIoFile::remoteIndexingCompleted()
 {
-    if (!indexed) {
-        ioRequestor->indexingCompleted();
-        indexed = true;
+    if (indexed) {
+        debugs(47, DBG_IMPORTANT, "BUG: a duplicated message from the already indexed disker" << diskId);
+        return;
     }
+
+    ioRequestor->remoteIndexingCompleted();
+    indexed = true;
```

Comment:
We should change state before notifying about the state changes.

```suggestion
    indexed = true;
    ioRequestor->remoteIndexingCompleted();
```